File: Python/121. Best Time to Buy and Sell Stock.py
from typing import List


# noinspection PyMethodMayBeStatic, PyPep8Naming
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minimum = float('inf')
        maximum = 0

        for price in prices:
            if price < minimum:
                minimum = price

            profit = price - minimum
            if profit > maximum:
                maximum = profit

        return maximum

File: Go/121. Best Time to Buy and Sell Stock.go
package main

//goland:noinspection GoUnusedFunction
func maxProfit(prices []int) int {
	minimum := 1<<31 - 1
	maximum := 0

	for _, price := range prices {
		if price < minimum {
			minimum = price
		}

		profit := price - minimum
		if profit > maximum {
			maximum = profit
		}
	}

	return maximum
}
